lib/csv2avro/converter.rb

```diff
@@ -35,11 +46,15 @@ def convert
           @writer.write(hash)
         rescue CSV2Avro::SchemaValidationError => e
           error_msg = "L#{row_number}: #{e.errors.join(', ')}"
-          @error_writer.puts(error_msg)
+          e.errors.each do |error|
+            logger.event('schema_violation', filename: @filename, line: row_number, cause: error)
```

Comment:
The more accepted term is `reason`, but of course we can always change that later. I'm not sure passing an exception is a good idea though, could we do `reason: error.message` instead?